src/Api/Controller/UploadLogoController.php

```diff
@@ -10,60 +10,35 @@
 namespace Flarum\Api\Controller;
 
 use Flarum\Settings\SettingsRepositoryInterface;
-use Illuminate\Support\Arr;
-use Illuminate\Support\Str;
+use Intervention\Image\Image;
 use Intervention\Image\ImageManager;
 use League\Flysystem\FilesystemInterface;
-use Psr\Http\Message\ServerRequestInterface;
-use Tobscure\JsonApi\Document;
+use Psr\Http\Message\UploadedFileInterface;
 
-class UploadLogoController extends ShowForumController
+class UploadLogoController extends UploadImageController
 {
     /**
-     * @var SettingsRepositoryInterface
-     */
-    protected $settings;
-
-    /**
-     * @var FilesystemInterface
-     */
-    protected $uploadDir;
-
-    /**
-     * @param SettingsRepositoryInterface $settings
-     * @param FilesystemInterface $uploadDir
+     * {@inheritdoc}
      */
     public function __construct(SettingsRepositoryInterface $settings, FilesystemInterface $uploadDir)
     {
-        $this->settings = $settings;
-        $this->uploadDir = $uploadDir;
+        parent::__construct($settings, $uploadDir);
+
+        $this->filenamePrefix = 'logo';
```

Comment:
Couldn't these go directly on the class?

Reply:
Of course they could, thank you for pointing this out @askvortsov1! I was concentrating on untangling the data() method and unwittingly wrote some unnecessary code myself. Now even more DRY, removing the constructors saves on some imports as well.